app/plugin.go

```diff
@@ -450,3 +455,98 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+// GetPluginPublicKeys returns all public keys listed in the config.
+func (a *App) GetPluginPublicKeys() ([]*model.PublicKeyDescription, *model.AppError) {
+	config := a.Config().PluginSettings
+
+	return config.SignaturePublicKeyFiles, nil
+}
+
+// GetPublicKey will return the actual public key saved in the `filename` file.
+func (a *App) GetPublicKey(filename string) ([]byte, *model.AppError) {
+	ext := filepath.Ext(filename)
+	if ext != PluginSignaturePublicKeyFileExtention {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.not_a_public_key.app_error", nil, "", http.StatusInternalServerError)
+	}
+	data, err := a.Srv.configStore.GetFile(filename)
+	if err != nil {
+		return nil, model.NewAppError("GetPublicKey", "api.plugin.get_public_key.get_file.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return data, nil
+}
+
+func (a *App) writePublicKeyFile(file string) *model.AppError {
```

Comment:
Agree on the convention
I think `App` object method should return `AppError`

Reply:
Agreed. However, even private methods? I say private/internal methods should return `error` and Public methods should definitely return `*model.AppError`. But in all honesty it doesn't make a difference because eventually everything will be using `error` 🎉 

I say private because i see similar convention already set https://github.com/mattermost/mattermost-server/blob/master/app/plugin_statuses.go#L73